app/src/main/java/com/rockthevote/grommet/ui/SessionViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package com.rockthevote.grommet.ui
+
+import androidx.lifecycle.*
+import com.rockthevote.grommet.data.db.dao.SessionDao
+import com.rockthevote.grommet.data.db.model.Session
+import com.rockthevote.grommet.util.coroutines.DispatcherProvider
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+class SessionViewModel(
+    private val sessionDao: SessionDao,
+    private val dispatchers: DispatcherProvider
+) : ViewModel() {
+
+    private val _state = MutableLiveData<SessionStartState>(SessionStartState.Init)
+    val state: LiveData<SessionStartState> = _state
+
+    fun updatePartnerId(newId: String) {
```

Comment:
The idea of this viewModel is to expose functions to update the state as the user goes through the flow(s) that create a session, appending the data to the state as it becomes available. This isn't all complete yet, but works in much the same way as the current Registration flow.